frontend/src/components/AddIngredients.js

```diff
@@ -3,20 +3,32 @@ import { TextField } from "@material-ui/core";
 import { Autocomplete } from "@material-ui/lab";
 import { useState } from "react";
 
-export default function AddIngredients({ ingredients, setIngredientsList }) {
+export default function AddIngredients({
+  ingredients,
+  ingredientsList,
+  setIngredientsList,
+}) {
   const [ingredient, setIngredient] = useState("");
+  const [description, setDescription] = useState("");
+  let checkForDouble = false;
 
   const handleAddClick = (event) => {
     event.preventDefault();
-    setIngredientsList((ingredientsList) => [
-      ...ingredientsList,
-      ingredient.name,
-    ]);
+    for (let i = 0; i < ingredientsList.length; i++) {
```

Comment:
I wonder if there is an array function that can `find`  an item based on some condition 🤔 Will greatly simplify your function. There really is hardly the need to write a for loop.

Reply:
Wanna change it with stream and find, but didn't work the last two times ^^ will fix it with future push.